pkgs/tools/networking/xrootd/test-runner.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, cppunit
+, xrootd
+}:
+
+xrootd.overrideAttrs (oldAttrs: {
+  passthru.tests = lib.removeAttrs "test-runner" (oldAttrs.passthru.tests or { });
+
+  cmakeFlags = (oldAttrs.cmakeFlags or [ ]) ++ [
+    "-DENABLE_TESTS=TRUE"
```

Comment:
Any reason to not always build with tests?

Reply:
`test-runner.nix` is meant to be imported as `passthru.tests.test-runner` by the main package.

I'm not sure if there would be some infinite recursion when it take itself as a passthru test.

Reply:
In `xrootd/default.nix`:
```Nix
passthru.tests.test-runner = callPackage ./test-runner.nix { };
```

Reply:
Let me rephrase. Why not just build the package once with the test enabled?

Reply:
The menu stated in section [installCheckPhase](https://nixos.org/manual/nixpkgs/stable/#ssec-installCheck-phase)

> It is often better to add tests that are not part of the source distribution to passthru.tests (see [Section 7.1.12, “tests”](https://nixos.org/manual/nixpkgs/stable/#var-meta-tests)). This avoids adding overhead to every build and enables us to run them independently.

Should I enable the test from the cmakeFlags of the original package by default, and just run the test without rebuilding in the testing package?

Reply:
> Should I enable the test from the cmakeFlags of the original package by default, and just run the test without rebuilding in the testing package?

Yes.